### PR TITLE
multi-tenant: Use unrestricted blob client for MT tests

### PR DESCRIPTION
**Pull request by ajstorm**
See individual commits for details.

Informs #76378.

Release note: None.

### CODE REVIEW

**Review by knz — APPROVED** [review]
> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/84200)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @renatolabs)

File: pkg/server/testserver.go [hunk 1]
```diff
@@ -817,11 +817,14 @@ func (ts *TestServer) StartTenant(
 	baseCfg.HeapProfileDirName = params.HeapProfileDirName
 	baseCfg.GoroutineDumpDirName = params.GoroutineDumpDirName
 
-	// TODO(ajstorm): Temporarily use a local-only blob client to get things
-	//  working for the common multi-tenant case. Will need to revisit this
-	//  to enable tests which require remote blob access. Tracked with #76378.
+	localNodeIDContainer := &base.NodeIDContainer{}
+	localNodeIDContainer.Set(context.Background(), ts.NodeID())
```
Comment (adityamaru) [inline pkg/server/testserver.go:821]: nit: can we just use the passed in `ctx` here?

**Review by adityamaru — APPROVED** [review]

File: pkg/server/testserver.go (hunk 1 above)
Comment (ajstorm) [reply pkg/server/testserver.go]: Done.  TFTR!

**Comment by ajstorm** [comment]
TFTRs!

bors r=knz,adityamaru

**Pull request merged**